src/unittest/test_servermodmanager.cpp

```diff
@@ -191,3 +193,84 @@ void TestServerModManager::testGetModMediaPaths()
 	sm.getModsMediaPaths(result);
 	UASSERTEQ(bool, result.empty(), false);
 }
+
+void TestServerModManager::testModDependencies()
+{
+	ModConfiguration mc(TEST_WORLDDIR);
+
+	/* Dependencies
+		NAME       HARD-DEPEND         OPT-DEPEND
+		default    (none)
+		dye        default
+		wool       default, dye
+		mesecons   default, invalid
+		pipeworks  default             mesecons
+		technic    pipeworks, mesecons
+		circ_1     default             circ_2
+		circ_2     circ_3
+		circ_3     circ_1
+
+		Expected:
+			mesecons:  unsatisfied dependencies (error)
+			pipeworks: opt-depend "mesecons" unsatisfied (ok)
+			technic:   unsatisfied dependencies (error)
+			circ_1:    circular reference. circ_2 not met
+	*/
+	std::vector<ModSpec> mods;
+	mods.emplace_back(ModSpec("default"));
+	mods.emplace_back(ModSpec("dye"));
+	mods[mods.size() - 1].depends.insert("default");
+
+	mods.emplace_back(ModSpec("wool"));
+	mods[mods.size() - 1].depends.insert("default");
+	mods[mods.size() - 1].depends.insert("dye");
+
+	mods.emplace_back(ModSpec("mesecons"));
+	mods[mods.size() - 1].depends.insert("default");
+	mods[mods.size() - 1].depends.insert("invalid"); // unknown mod
+
+	mods.emplace_back(ModSpec("pipeworks"));
+	mods[mods.size() - 1].depends.insert("default");
+	mods[mods.size() - 1].optdepends.insert("mesecons");
+
+	mods.emplace_back(ModSpec("technic"));
+	mods[mods.size() - 1].depends.insert("pipeworks");
+	mods[mods.size() - 1].depends.insert("mesecons");
+
+	mods.emplace_back(ModSpec("circ_1"));
+	mods[mods.size() - 1].depends.insert("default");
+	mods[mods.size() - 1].optdepends.insert("circ_2");
+
+	mods.emplace_back(ModSpec("circ_2"));
+	mods[mods.size() - 1].depends.insert("circ_3");
+	mods.emplace_back(ModSpec("circ_3"));
+	mods[mods.size() - 1].depends.insert("circ_1");
+
```

Comment:
You can use `mods.back()` instead of `mods[mods.size() - 1]` everywhere here.

Reply:
(Done)